crates/primitives/snos-output/src/lib.rs

```diff
@@ -0,0 +1,59 @@
+//! StarkNet OS program output primitives.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[doc(hidden)]
+pub extern crate alloc;
+
+mod codec;
+mod conversions;
+mod felt_reader;
+
+#[cfg(test)]
+mod tests;
+
+use alloc::vec::Vec;
+
+use starknet_api::hash::StarkFelt;
+
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+#[cfg_attr(feature = "scale-info", derive(scale_info::TypeInfo))]
+#[cfg_attr(feature = "parity-scale-codec", derive(parity_scale_codec::Encode, parity_scale_codec::Decode))]
+/// Main part of Starknet OS program output
+pub struct StarknetOsOutput {
+    /// The state commitment before this block.
+    pub prev_state_root: StarkFelt,
+    /// The state commitment after this block.
+    pub new_state_root: StarkFelt,
+    /// The number (height) of this block.
+    pub block_number: StarkFelt,
+    /// The hash of this block.
+    pub block_hash: StarkFelt,
+    /// The Starknet chain config hash
+    pub config_hash: StarkFelt,
+    /// List of messages sent to L1 in this block
+    pub messages_to_l1: Vec<MessageL2ToL1>,
+    /// List of messages from L1 handled in this block
+    pub messages_to_l2: Vec<MessageL1ToL2>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+#[cfg_attr(feature = "scale-info", derive(scale_info::TypeInfo))]
+#[cfg_attr(feature = "parity-scale-codec", derive(parity_scale_codec::Encode, parity_scale_codec::Decode))]
+/// Message sent to L1 by invoking according Statknet syscall
+pub struct MessageL2ToL1 {
```

Comment:
We are actually using a similar [type](https://github.com/apoorvsadana/madara/blob/14b0b62b1155feb99ba321030f6ec7bd2397391b/crates/primitives/simulations/src/lib.rs#L118) for simulate tx as well. The blockifier has its own type for `MessageToL1` and `MessageToL2`, so instead of mapping the blockifier types to starknet api, I created the struct within the primitives so we have more control (like we did for transactions). If we want to continue with this, should we move this type to maybe the transactions primitive and reuse it for simulate txs?

Reply:
We can share the types when both PR are merged